src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,48 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+/**
+ * Adapter from Job2dDriver to DrawPanelController
+ *
+ */
+public class LineDrawerAdapter implements Job2dDriver {
+    private int startX = 0, startY = 0;
+    private final DrawPanelController drawPanelController;
+
+    public LineDrawerAdapter(DrawPanelController controller) {
+        if (controller == null) {
+            throw new IllegalArgumentException("Provided controller must not be null");
+        }
+        drawPanelController = controller;
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line;
+        if (DrawerFeature.isSpecialLineActive()) {
+            line = LineFactory.getSpecialLine();
+        } else {
+            line = LineFactory.getBasicLine();
+        }
```

Comment:
Hardcoded line type. What about 3.3.2 test?
Direct coupling with LineFactory class is unnecessary.